### PR TITLE
Add privacy policy link page

### PR DESCRIPTION
**Pull request by DavidBiddle**
#### What problem does the pull request solve?
Allows a user to set their form's privacy policy URL.

Trello card: https://trello.com/c/nmMtizUC/365-add-provide-a-link-to-your-privacy-information-to-forms-admin-needs-dev-review

#### Checklist

- [x] I've used the pull request template
- [x] I've linked this PR to the relevant issue (if mission work)
- [x] I've written unit tests for these changes (if code change)
- [n/a] I've updated the documentation in (If any documentation requires updating)
    - [n/a] README.md
    - [n/a] Elsewhere (please link)

### CODE REVIEW

File: app/controllers/forms/privacy_policy_controller.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+module Forms
+  class PrivacyPolicyController < BaseController
+    def new
+      @privacy_policy_form = PrivacyPolicyForm.new(form: current_form).assign_form_values
+    end
+
+    def create
+      @privacy_policy_form = PrivacyPolicyForm.new(privacy_policy_form_params(current_form))
```
Comment (aliuk2012) [inline app/controllers/forms/privacy_policy_controller.rb:8]:
```suggestion
      @privacy_policy_form = PrivacyPolicyForm.new(privacy_policy_form_params)
```

File: app/controllers/forms/privacy_policy_controller.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        redirect_to form_path(@privacy_policy_form.form)
+      else
+        render :new
+      end
+    end
+
+  private
+
+    def privacy_policy_form_params(form)
+      params.require(:forms_privacy_policy_form).permit(:privacy_policy_url).merge(form:)
```
Comment (aliuk2012) [inline app/controllers/forms/privacy_policy_controller.rb:20]:
```suggestion
    def privacy_policy_form_params
      params.require(:forms_privacy_policy_form).permit(:privacy_policy_url).merge(current_form)
```

File: spec/forms/privacy_policy_form_spec.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    it "is invalid if URL is not in the correct format" do
+      privacy_policy_form = described_class.new(privacy_policy_url: "laura.mipsum")
+      error_message = I18n.t("activemodel.errors.models.forms/privacy_policy_form.attributes.privacy_policy_url.invalid_url")
+
+      privacy_policy_form.validate(:privacy_policy_url)
+
+      expect(privacy_policy_form.errors.full_messages_for(:privacy_policy_url)).to include(
+        "Privacy policy url #{error_message}",
+      )
+    end
```
Comment (aliuk2012) [inline spec/forms/privacy_policy_form_spec.rb:25]: Could replace this with a single line 

https://github.com/perfectline/validates_url#with-rspec

File: config/locales/forms/privacy_policy.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+        privacy_policy_url: |-
+          For example, https://www.gov.uk/help/privacy-notice
+  activemodel:
+    errors:
+      models:
+        forms/privacy_policy_form:
+          attributes:
+            privacy_policy_url:
+              blank: Enter a link to privacy information
+              invalid_url: Enter an link in the correct format, like https://www.gov.uk/help/privacy-notice
```
Comment (aliuk2012) [inline config/locales/forms/privacy_policy.yml:17]:
```suggestion
              invalid_url: Enter a link in the correct format, like https://www.gov.uk/help/privacy-notice
```

**Review by aliuk2012 — COMMENTED** [review]
A couple comments but otherwise looks great

File: spec/forms/privacy_policy_form_spec.rb (hunk 3 above)
Comment (aliuk2012) [reply spec/forms/privacy_policy_form_spec.rb:25]: No problem, I can pick this up.

Comment (aliuk2012) [reply spec/forms/privacy_policy_form_spec.rb]: [0a2e743](https://github.com/alphagov/forms-admin/pull/108/commits/0a2e7430f43c83c9b5715faad8dc66665df69e94)

**Review by thomasiles — APPROVED** [review]

**Review by aliuk2012 — APPROVED** [review]

**Pull request merged**